File: code/export_messages_to_csv/ListMessages.php
<?php
require './vendor/autoload.php';
use SignalWire\Rest\Client;
$client = new Client('ProjectID', 'AuthToken', array("signalwireSpaceUrl" => "YOURSPACE.signalwire.com"));
// filters by whatever parameters you need, simply uncomment to enable more or comment to enable less
$messages = $client->messages->read([
    "dateSentAfter" => "2021-04-01",
    "dateSentBefore" => "2021-04-20",
    //'Status' => 'busy', // filter by Status
    //'From' => '+1xxxxxxxxxx', // filter by From
    //'To' => '+1xxxxxxxxxx', // filter by To
]);
// Write Headers
$fields = array('Message SID', 'From', 'To', 'Date Sent', 'Status', 'Direction', 'Price');
echo '"'.implode('","', $fields).'"'."\n";
// Open File named TodaysDate_messageReport
$fp = fopen(date("Y-m-d").'_messageReport.csv', 'w');
// Insert headers
fputcsv($fp, $fields);
// Write rows
foreach ($messages as $message) {
    $row = array(
        $message->sid, 
        $message->from, 
        $message->to, 
        $message->dateSent->format('Y-m-d H:i:s'), 
        $message->status, 
        $message->direction, 
        $message->price,
    );
    // Insert rows into CSV
    fputcsv($fp, $row);
    echo '"'.implode('","', $row).'"'."\n";
}
// close file
fclose($fp);

File: code/export_messages_to_csv/ListMessages.py
# This code snippet is to list messages from your LaML logs filtered by from number
# It will then put the data into a dataframe and export to CSV.
# You MUST have pandas and datetime installed and imported for this to work.
from datetime import datetime
from signalwire.rest import Client as signalwire_client
import pandas as pd

client = signalwire_client("ProjectID", "AuthToken", signalwire_space_url = 'SpaceUrl')


# Lists messages from a particular number within a specific date range and from a number
messages = client.messages.list(date_sent_after=datetime(2021, 1, 1, 0, 0),
                               # date_sent_before=datetime(2021, 3, 1, 0, 0),
                               from_='+15552308945',
                               )

# Sets up an empty array
d = []

# Appends all data from messages into an array
for record in messages:
   d.append((record.from_, record.to, record.date_sent, record.status, record.sid))

print(d)

# Puts message log array into dataframe with headers for easier reading.
df = pd.DataFrame(d, columns=('From', 'To', 'Date', 'Status', 'MessageSID'))

print('dataframe')
print('\n')
print(df)

# Exports dataframe to csv, index=False turns off the indexing for each row
df.to_csv('messages.csv', index=False, encoding='utf-8')
